### PR TITLE
Feature/api end points

### PR DESCRIPTION
**Pull request by saied2035**
### Project requirements

- I create an API endpoint to list all posts for a user.
- I create an API endpoint to list all comments for a user's post.
- I create an API endpoint to add a comment to a post.
- I add authentication to your API endpoints.
- The API endpoints should receive JSON and respond in JSON as well.

Thanks for reviewing the PR 😄

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,6 +8,16 @@
...
   end
+
+  namespace :api do
+    resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :new,:create,:show,:destroy] do
+      resources :comments, only: [:index, :new, :create, :destroy]
+      resources :likes, only: [:create]
+    end
+  end    
+ end
```
Comment (descholar-ceo) [inline config/routes.rb:19]: These endpoints don't create the following APIs as required in requirements: 

1. Has API endpoint that returns all post in JSON format - **GET :** `/api/users/:user_id/posts` or `/api/v1/users/:user_id/posts`
1. Has API endpoint that returns all comments for a post - **GET :** `/users/:user_id/posts/:id/comments`
1. Has API endpoint that adds' a comment to a post - **POST :** `/users/:user_id/posts/:id/comments`

Try to put these api routes inside user

**Review by descholar-ceo — APPROVED** [review]
Hi @saied2035 && @RayhanTabase , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
👍🏽 All API endpoints receives JSON and responds in a JSON format as well. ✔️ 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (saied2035) [reply config/routes.rb:19]: @descholar-ceo  Thanks for reviewing our Pr. I fixed it now 😄

**Review by EricMbouwe — APPROVED** [review]
Hi @saied2035, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**